### PR TITLE
Add coverage measurement with gcov

### PR DESCRIPTION
**Pull request by mame**
Usage: `./configure --enable-gcov && make && make exam && make gcov`

### CODE REVIEW

**Comment by mame** [comment]
@nobu Could you review this?  I'm unsure about build scripts.  I don't know the difference of `cflags` and `CFLAGS`.

File: tool/run-gcov.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+require "open3"
+
+BASE_DIR = Pathname(__dir__).dirname
+
+# Hack to allow gcov to find parse.c
+unless File.symlink?("ext/ripper/parse.c")
+  File.symlink("../../parse.c", "ext/ripper/parse.c")
+end
+
+Pathname.glob(BASE_DIR + "**/*.gcda").sort.each do |gcda|
```
Comment (nobu) [inline tool/run-gcov.rb]: Are `*.gcda` files generated in the source directory?

File: configure.in [hunk 2]
```diff
@@ -4063,6 +4063,15 @@ AC_SUBST(DTRACE_REBUILD)
 AC_SUBST(DTRACE_OPT)
 AC_SUBST(LIBRUBY_A_OBJS)
 
+AC_ARG_ENABLE(gcov,
+       AS_HELP_STRING([--enable-gcov], [enable coverage measurement by gcov]),
+       [AC_SUBST(gcov, yes)])
```
Comment (nobu) [inline configure.in]: Or, add `$(gcov)` to `CFLAGS` and `LDFLAGS` in Makefile.in?

**Comment by mame** [comment]
@nobu thank you!  I applied your comments.  I'll commit this later.

**Comment by mame** [comment]
Committed at r58817.

**Pull request closed without merging**